internal/pkg/manifest/validate_env.go

```diff
@@ -96,27 +95,23 @@ func (cfg SecurityGroupRule) validate() error {
 			missingField: "ip_protocol",
 		}
 	}
-	return cfg.PortsConfig.validate()
+	return cfg.Ports.validate()
 }
 
 // validate if ports are set.
-func (cfg PortsConfig) validate() error {
+func (cfg portsConfig) validate() error {
 	if cfg.IsEmpty() {
 		return &errFieldMustBeSpecified{
 			missingField: "ports",
 		}
 	}
-	if !cfg.Ports.IsEmpty() {
-		return cfg.Ports.validate()
-	}
-	return nil
-}
-
-// validate checks if PortsRangeBand is set correctly.
-func (str PortsRangeBand) validate() error {
-	ports := strings.Split(string(str), "-")
-	if len(ports) != 2 {
-		return fmt.Errorf("invalid ports value %s: valid port format is ${from_port}-${to_port}", string(str))
+	if cfg.Ports != nil {
+		if err := cfg.Ports.validate(); err != nil {
+			if strings.Contains(err.Error(), "invalid range value") {
+				return fmt.Errorf("invalid ports value %s: valid port format is ${from_port}-${to_port}", *cfg.Ports)
+			}
+			return err
+		}
```

Comment:
nit: what do you think of flipping the conditional
```suggestion
        if cfg.Ports == nil {
           retur nil
        }
	// check `cfg.Ports.validate()`
}
```

Reply:
Another nit that I'd have is that we could think of a more robust way to match the invalid range value error. Consider if one day we decide to change the error message for `IntRangeBand`'s `validate`, from `"invalid range value..."` to  something like`"range %s is invalid"`, it's likely that we'd forget to update it here 😰 . 

It is not entirely necessary for us to address the nit here. Just sharing the ⬇️ code snippets for your thoughts 💭 

An alternative that can deal with this & does not involve too much modification on top of what we have here is:
```go
// In validate.go
type errInvalidRange struct {
	value       string
	validFormat string
}

func (e *errInvalidRange) Error() string {
	return fmt.Sprintf("invalid range value %s: valid format is %s", e.value, e.validFormat)
}

// validate returns nil if IntRangeBand is configured correctly.
func (r IntRangeBand) validate() error {
	....
	// Valid minMax example: ["1-2", "1", "2"]
	if len(minMax) != 3 {
		return &errInvalidRange{
			value:       str,
			validFormat: "${min}-${max}",
		}
	}
        ...
}
```
then here:
```go
// In validate_env.go
func (cfg portsConfig) validate() error {
  ...
 if err := cfg.Ports.validate(); err != nil {
      var targetErr *errInvalidRange
      if errors.As(err, &targetErr) {
	      return &errInvalidRange{
		      value:       aws.StringValue((*string)(cfg.Ports)),
		      validFormat: "${from_port}-${to_port}",
	      }
      }
      return err
    }
    return nil
}
```

Reply:
Hmm totally make sense to me. We need robust way to handle such kinds of errors and your solution looks good to me. I will make changes in this PR itself to add this robust change now only in copilot.